Bindings/Java/Matlab/Dynamic_Walking_Challenge/Dynamic_Walker_Simulation/walker_simulation_objective_function.m

```diff
@@ -0,0 +1,69 @@
+function J = walker_simulation_objective_function(vod)
+%% walker_simulation_objective_function.m
+% Function for running forward simulation of a walker model and computing 
+% the objective function (Pelvis X translation). The input to this function 
+% is a 12x1 vector of initial coordinate value and speed states for a 
+% dynamic walker model. See Matlab script walkerOptimization.m for use.  
+
+% Written by James Dunne, Carmchael Ong, Tom Uchida, Ajay Seth. 
+
+% Define Global variables. 
+global model translation_store vectorOfDefaults_store;
+
+% Import OpenSim libraries 
+import org.opensim.modeling.*
+
+% Store the vector of defaults that the optimizer chooses
+vectorOfDefaults_store = [vectorOfDefaults_store;vod']; 
+
+% Set the time for the simulation to take place  
+finalTime = 10;
+
+% Update model default coordinate values and speeds with optimizer defined 
+% values.
+model.getCoordinateSet().get('Pelvis_tx').setDefaultValue(vod(1));
+model.getCoordinateSet().get('Pelvis_tx').setDefaultSpeedValue(vod(2));
+model.getCoordinateSet().get('Pelvis_ty').setDefaultValue(vod(3));
+model.getCoordinateSet().get('Pelvis_ty').setDefaultSpeedValue(vod(4));
+model.getCoordinateSet().get('LHip_rz').setDefaultValue(vod(5));
+model.getCoordinateSet().get('LHip_rz').setDefaultSpeedValue(vod(6));
+model.getCoordinateSet().get('RHip_rz').setDefaultValue(vod(7));
+model.getCoordinateSet().get('RHip_rz').setDefaultSpeedValue(vod(8));
+model.getCoordinateSet().get('LKnee_rz').setDefaultValue(vod(9));
+model.getCoordinateSet().get('LKnee_rz').setDefaultSpeedValue(vod(10));
+model.getCoordinateSet().get('RKnee_rz').setDefaultValue(vod(11));
+model.getCoordinateSet().get('RKnee_rz').setDefaultSpeedValue(vod(12));
+
+% Initialize the model and get the state
+s = model.initSystem();
```

Comment:
How long does it take to run the whole optimization? I would expect this line to be relatively expensive. If you're just optimizing the initial conditions (i.e., the initial value for each state variable) and not the structure of the computational system itself (e.g., adding/removing bodies), then is it not possible to call `initSystem()` _once_ in walkerOptimization.m and use `vod` in walker_simulation_objective_function.m to adjust initial conditions rather than defaults?